intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,772 @@
+   "metadata": {},
+   "source": [
+    "#### The cell below runs the workflow end to end including the ETL and XGBoost\n",
+    "\n",
+    "**Note:** The mortgage dataset for years 2000-2016 is about 200GB of data. The total GPU memory and the max memory per gpu are two key parameters that determine the amount of data the system would be able to process for the workflow. The `determine_dataset` utility used below automatically queries these two parameters on the machine running the script and decides suitable values `start_year`, `end_year`, `part_count` and variation of the dataset (1GB split part files vs 2GB split part files) that should work on such systems.\n",
+    "\n",
+    "However, if you'd like to use existing data that has already been downloaded to your own location, or manually adjust these parameters based on the amount of data needed for processing, you can change these parameters provided in the notebook, by assigning new values to the variables or setting enivronment variables for `MORTGAGE_DATA_DIR` and `PART_COUNT`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if __name__ == \"__main__\":\n",
+    "\n",
+    "    import cudf\n",
+    "    import xgboost as xgb\n",
+    "    import dask_cudf\n",
+    "\n",
+    "    cmd = \"hostname --all-ip-addresses\"\n",
+    "    process = subprocess.Popen(cmd.split(), stdout=subprocess.PIPE)\n",
+    "    output, error = process.communicate()\n",
+    "    IPADDR = str(output.decode()).split()[0]\n",
+    "\n",
+    "    cluster = LocalCUDACluster(ip=IPADDR)\n",
+    "    client = Client(cluster)\n",
+    "\n",
+    "    data_dir = os.environ.get(\"MORTGAGE_DATA_DIR\", \"\") # Default to current working directory\n",
+    "    res = client.run(memory_info)\n",
+    "    # Total GPU memory on the system\n",
+    "    total_mem = sum(res.values()) \n",
+    "    # Memory of a single GPU on the machine\n",
+    "    # If the machine has multiple GPUs of different sizes, this is the size of the smallest GPU\n",
+    "    min_mem = min(res.values()) \n",
+    " \n",
+    "    # Start year for processing mortgage data\n",
+    "    start_year = None\n",
+    "    # End year for processing mortgage data\n",
+    "    end_year = None\n",
+    "    # The number of part files to train against. \n",
+    "    # If not provided, default to auto selection based on GPU memory available on the system\n",
+    "    part_count = os.environ.get(\"part_count\")\n",
+    "\n",
+    "    start_year, end_year, part_count, use_1GB_splits = determine_dataset(\n",
+    "        total_mem=total_mem, min_mem=min_mem, part_count=part_count\n",
+    "    )\n",
+    "\n",
+    "    # Download data based on these parameters\n",
+    "    # The 2GB split mortgage performance files are used if the system has 32GB GPUs.\n",
+    "    # On machines with GPUs less than 32GB we use the 1GB split files (to help reduce memory load)\n",
```

Comment:
Reading this, it would be nice to have an additional statement about general use of larger partitions/splits gives better performance, with the tradeoff of chunkier memory usage that 32GB cards can handle but 16GB cards cannot.

That's a complex discussion that I feel warrants its own paragraph in the markdown cell above, as opposed to just code comments.

Reply:
Reorganized the structure of the cell to describe the two system configurations `total mem` and `single gpu mem` and the impact it has on the parameters for the mortgage notebook.